### PR TITLE
🐛 (go/v4): Start to check ENABLE_WEBHOOKS envvar into main.go to fix issue when webhooks are used

### LINKED ISSUE #3559: Book ERROR   setup   problem running manager {"error": "open /tmp/k8s-webhook-server/serving-certs/tls.crt: no such file or directory"}
### What broke? What's expected?

Following the book tutorial in ordered steps, when getting to this page https://book.kubebuilder.io/cronjob-tutorial/running

At this point of the tutorial the user doesn't want to run the webhooks, just test with `make run` and disable the webhooks.

It intructs the user to skip the webhook with environment variable
```shell
export ENABLE_WEBHOOKS=false
make run
```
I expect to not see any errors.
The previous page https://book.kubebuilder.io/cronjob-tutorial/webhook-implementation
Doesn't mention at all that the main.go needs be modified from:
```go
	if err = (&batchv1.CronJob{}).SetupWebhookWithManager(mgr); err != nil {
		setupLog.Error(err, "unable to create webhook", "webhook", "CronJob")
		os.Exit(1)
	}
```
Modified to:
```go
	if os.Getenv("ENABLE_WEBHOOKS") != "false" {
		if err = (&batchv1.CronJob{}).SetupWebhookWithManager(mgr); err != nil {
			setupLog.Error(err, "unable to create webhook", "webhook", "CronJob")
			os.Exit(1)
		}
	}
```

### Reproducing this issue

Follow tutorial in order up to this page Following the book tutorial in ordered steps, when getting to this page https://book.kubebuilder.io/cronjob-tutorial/running

It intructs the user to skip the webhook with environment variable
```shell
export ENABLE_WEBHOOKS=false
make run
```

When running I get the following error 
```
2023-08-25T17:57:51Z    ERROR   setup   problem running manager {"error": "open /tmp/k8s-webhook-server/serving-certs/tls.crt: no such file or directory"}
main.main
        /workspaces/kubebuilder-book-club-aws/cmd/main.go:116
runtime.main
        /usr/local/go/src/runtime/proc.go:250
```

### KubeBuilder (CLI) Version

3.11.1

### PROJECT version

3

### Plugin versions

_No response_

### Other versions

_No response_

### Extra Labels

/kind documentation

### PR DESCRIPTION
**Pull request by lowang-bh**
Fixes #3559

describe: fix issue that webhook generated code in cmd/main.go ignored checking the env ENABLE_WEBHOOKS

Test result :
```
➜  guestbook git:(dev) ✗ export ENABLE_WEBHOOKS=false; make run
test -s /Users/wlh/go/src/github.com/lowang-bh/demo/guestbook/bin/controller-gen && /Users/wlh/go/src/github.com/lowang-bh/demo/guestbook/bin/controller-gen --version | grep -q v0.13.0 || \
	GOBIN=/Users/wlh/go/src/github.com/lowang-bh/demo/guestbook/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.13.0
/Users/wlh/go/src/github.com/lowang-bh/demo/guestbook/bin/controller-gen rbac:roleName=manager-role crd webhook paths="./..." output:crd:artifacts:config=config/crd/bases
/Users/wlh/go/src/github.com/lowang-bh/demo/guestbook/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
go vet ./...
go run ./cmd/main.go
2023-09-02T18:47:24+08:00	INFO	setup	starting manager
2023-09-02T18:47:24+08:00	INFO	starting server	{"kind": "health probe", "addr": "[::]:8081"}
2023-09-02T18:47:24+08:00	INFO	controller-runtime.metrics	Starting metrics server
2023-09-02T18:47:24+08:00	INFO	controller-runtime.metrics	Serving metrics server	{"bindAddress": ":8080", "secure": false}
2023-09-02T18:47:24+08:00	INFO	Starting EventSource	{"controller": "guestbook", "controllerGroup": "webapp.my.domain", "controllerKind": "Guestbook", "source": "kind source: *v1.Guestbook"}
2023-09-02T18:47:24+08:00	INFO	Starting Controller	{"controller": "guestbook", "controllerGroup": "webapp.my.domain", "controllerKind": "Guestbook"}
2023-09-02T18:47:24+08:00	INFO	Starting workers	{"controller": "guestbook", "controllerGroup": "webapp.my.domain", "controllerKind": "Guestbook", "worker count": 1}
```

### CODE REVIEW

**Comment by lowang-bh** [comment]
Test finished successfully
image

File: pkg/plugins/golang/v2/scaffolds/internal/templates/main.go [hunk 1]
```diff
@@ -114,9 +114,11 @@ const (
...
 	}
 `
-	webhookSetupCodeFragment = `if err = (&%s.%s{}).SetupWebhookWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create webhook", "webhook", "%s")
-		os.Exit(1)
+	webhookSetupCodeFragment = `if os.Getenv("ENABLE_WEBHOOKS") != "false" {
+		if err = (&%s.%s{}).SetupWebhookWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create webhook", "webhook", "%s")
+			os.Exit(1)
+		}
```
Comment (camilamacedo86) [inline pkg/plugins/golang/v2/scaffolds/internal/templates/main.go:121]: This plugin is deprecated for. along period so we no longer add changes on it. 
Could you please remove the change here?

File: pkg/plugins/golang/v3/scaffolds/internal/templates/main.go [hunk 2]
```diff
@@ -113,9 +113,11 @@ const (
...
 	}
 `
-	webhookSetupCodeFragment = `if err = (&%s.%s{}).SetupWebhookWithManager(mgr); err != nil {
-		setupLog.Error(err, "unable to create webhook", "webhook", "%s")
-		os.Exit(1)
+	webhookSetupCodeFragment = `if os.Getenv("ENABLE_WEBHOOKS") != "false" {
+		if err = (&%s.%s{}).SetupWebhookWithManager(mgr); err != nil {
+			setupLog.Error(err, "unable to create webhook", "webhook", "%s")
+			os.Exit(1)
+		}
```
Comment (camilamacedo86) [inline pkg/plugins/golang/v3/scaffolds/internal/templates/main.go:120]: This plugin (go/v3) is also deprecated. 
Could you please remove the change here?

**Review by camilamacedo86 — CHANGES_REQUESTED** [review]
Thank you for your contribution 🥇 

Following some nits

- The plugins go/v2 and go/v3 are deprecated so changes should not be performed on them.
- After change the scaffold you need to run `make generate` so that all samples under testdata and docs will be properly updated. Could you please to do it?

File: pkg/plugins/golang/v3/scaffolds/internal/templates/main.go (hunk 2 above)
Comment (lowang-bh) [reply pkg/plugins/golang/v3/scaffolds/internal/templates/main.go]: removed.

File: pkg/plugins/golang/v2/scaffolds/internal/templates/main.go (hunk 1 above)
Comment (lowang-bh) [reply pkg/plugins/golang/v2/scaffolds/internal/templates/main.go]: removed

**Comment by lowang-bh** [comment]
> * After change the scaffold you need to run `make generate`

Done.

image

**Comment by camilamacedo86** [comment]
@lowang-bh,

Thank you.
Could you please squash the commits?
We need to have always just one commit to get merged per PR


/ok-to-test

**Comment by lowang-bh** [comment]
> Could you please squash the commits?

Done.

> I also updated title for we have a better info when we generate the release notes. I hope that you do not mind

Thanks.

**Review by camilamacedo86 — APPROVED** [review]
/lgtm
/approved

**Comment by camilamacedo86** [comment]
@lowang-bh 

You need to run make generate =, see : https://github.com/kubernetes-sigs/kubebuilder/actions/runs/6063945828/job/16452451257?pr=3578#step:6:47

/hold

**Comment by lowang-bh** [comment]
> 

Yes, I am looking forward this issue. It always insert two pices of same code.

**Review by camilamacedo86 — COMMENTED** [review]
/lgtm

**Comment by camilamacedo86** [comment]
The API fails because the change in the doc see:

sigs.k8s.io/kubebuilder/v3/hack/docs/internal/cronjob-tutorial
  Incompatible changes:
  - MainEnableWebhook: value changed from "\n\n\t/*\n\t\tWe'\''ll also set up webhooks for our type, which we'\''ll t... to "\n\n\t/*\n\t\tWe'\''ll also set up webhooks for our type, which we'\''ll t...'
+ GOAPIDIFF_RC=1

Therefore, it is not a genuine breakchange.
So, we can move forward with this one.

**Pull request merged**